src/ert/analysis/_es_update.py

```diff
@@ -537,15 +516,34 @@ def analysis_ES(
                 f"No active observations for update step: {update_step.name}."
             )
 
-        smoother = ies.ES()
+        inversion_types = {0: "exact", 1: "subspace", 2: "subspace", 3: "subspace"}
+        inversion_type = inversion_types.get(module.ies_inversion, "Invalid input")
+        smoother_es = ies.ESMDA(
+            covariance=observation_errors**2,
+            observations=observation_values,
+            alpha=1,  # The user is responsible to scale observation covariance (esmda usage)
+            seed=rng,
+            inversion=inversion_type,
+        )
         truncation = module.enkf_truncation
-        noise = rng.standard_normal(size=(num_obs, ensemble_size))
 
         if module.localization:
-            Y_prime = S - S.mean(axis=1, keepdims=True)
-            Sigma_Y = np.std(S, axis=1, ddof=1)
             batch_size: int = 1000
-            correlation_threshold = module.correlation_threshold(ensemble_size)
+            smoother_adaptive_es = AdaptiveESMDA(
+                covariance=observation_errors**2,
+                observations=observation_values,
+                seed=rng,
+            )
+
+            # Pre-calculate cov_YY
```

Comment:
I would consider using `empirical_cross_covariance` from `esmda_inversion` here. Just to keep all linear algebra and computations in the package.

Reply:
Or just `np.cov(S)` perhaps? Considering it is ordinary sample covariance and not cross-covariance? But indeed this linear algebra should be outsourced